service/history/queuev2/queue_base.go

```diff
@@ -274,6 +274,14 @@ func (q *queueBase) processNewTasks() bool {
 	return true
 }
 
+func (q *queueBase) insertSingleTask(task task.Task) bool {
+	return q.virtualQueueManager.InsertSingleTaskToRootQueue(task)
```

Comment:
if we only insert task to root queue, there is a risk of losing task if the task doesn't belong to root queue

Reply:
Haven't thought about it yet. The rough idea is to find the virtual slice includes the task and its corresponding virtual queue.